modules/web/src/app/node-data/component.ts

```diff
@@ -260,8 +293,28 @@ export class NodeDataComponent extends BaseFormValidator implements OnInit, OnDe
         this.dialogEditMode || this.isCusterTemplateEditMode
           ? this._nodeDataService.nodeData.count
           : settings.defaultNodeCount;
+      this._settings = settings;
       this.form.get(Controls.Count).setValue(replicas);
     });
+
+    this._clusterSpecService.providerChanges
```

Comment:
We already have `providerChanges` subscription so can we move this code to that subscription?